### DISCUSSION TITLE
Disable DropdownList

### DISCUSSION THREAD

**Discussion by marcofo23** [Megabit/Blazorise#2387] [Q&A] [answered]
Hi!
I'm new with Blazor so forgive me if the question is stupid.

I'm trying to use the DropdownList component and I want to disable it, how can I do this ?
I tried with Attributes parameter but It seems not working.

Thanks

**Comment by stsrki** [comment, marked as answer]
Don't worry we will not judge. If you don't ask how else you will learn :)

---

I checked and It seems I forgot to include the `Disabled` parameter on a `DropdownList`. Unfortunately, until I fix it your only option is to build a dropdown menu from scratch using the `Dropdown` component(s).

DropodownList is internally very simple, you should be able to make it without any problems:

```cshtml
<Dropdown @ref="@dropdownRef" ElementId="@ElementId" Class="@Class" Style="@Style" Attributes="@Attributes">
    <DropdownToggle @ref="@dropdownToggleRef" Color="@Color" TabIndex="@TabIndex">@ChildContent</DropdownToggle>
    <DropdownMenu>
        @if ( Data != null )
        {
            foreach ( var item in Data )
            {
                var text = TextField?.Invoke( item );
                var value = ValueField != null ? ValueField.Invoke( item ) : default;

                <DropdownItem Clicked="@HandleDropdownItemClicked" Value="@value">@text</DropdownItem>
            }
        }
    </DropdownMenu>
</Dropdown>
```

**Reply by marcofo23** [reply]
No problem, I'm just doing some tests to learn so I'm not in a hurry :)
Thanks

**Reply by marcofo23** [reply]
PS: let us know when is fixed ! ;)

**Reply by stsrki** [reply]
I'm doing it now, but it will not be available until the next 0.9.4 preview.

**Reply by stsrki** [reply]
Done #2388

**Reply by marcofo23** [reply]
Same problem regarding SelectList extension, right ?

**Reply by stsrki** [reply]
SelectList has a Disabled parameter as far as I can see.

**Reply by marcofo23** [reply]
I can't find it.

I searched here
https://blazorise.com/docs/extensions/selectlist/
and in the source code that I've downloaded (https://github.com/Megabit/Blazorise/blob/master/Source/Extensions/Blazorise.Components/SelectList.razor.cs).

Where am i wrong?

**Reply by stsrki** [reply]
Sorry about that :)

It is done recently, and is also only for v0.9.4.

**Reply by marcofo23** [reply]
Ah ok, I was worrying about myself :)
No problem 👍
